### DISCUSSION TITLE
Exclusion Environment

### DISCUSSION THREAD

**Discussion by LynetteCullens** [artisticat1/obsidian-latex-suite#252] [Q&A] [answered]
How do I exclude fractions from \pu environment?`[
		["^{", "}"],
		["\\pu{", "}"],
		["\pu{", "}"]
	]`
That didn't work. Also, `[
		["^{", "}"],
		["\\pu{", "}"]
	]` didn't work. 
Another question is how do I prevent \\pu from outputting when I'm trying to type \pu? I know how to remove the snippet from the default settings, but am I using the plugin incorrectly?

**Comment by artisticat1** [comment, marked as answer]
> How do I exclude fractions from \pu environment?

This is already done by default; here is the default setting for reference:

```
[
	["^{", "}"],
	["\\pu{", "}"]
]
```

> Another question is how do I prevent \pu from outputting when I'm trying to type \pu? I know how to remove the snippet from the default settings, but am I using the plugin incorrectly?

Yes, simply remove the snippet in the plugin settings. You can customise any/all of the snippets to your liking; there's no "incorrect" way to use the plugin!

**Reply by LynetteCullens** [reply]
![pu environment](https://github.com/artisticat1/obsidian-latex-suite/assets/45722611/ed4b75e6-2cb1-4984-91dc-2afc6cd6e9b2)
My current Excluded environments have been reset to default.
`[
	["^{", "}"],
	["\\pu{", "}"]
]`
Also, I have been using the plugin incorrectly by typing "\pu" instead of just "pu", I kept getting "\\pu".

Edit: I've figured it out thank you. I didn't realize there was also an autocomplete fraction function inside the default snippet file that went along with the auto-fraction toggle function. 
I've just edited it to read:
`	{trigger: "//", replacement: "\\frac{$0}{$1}$2", options: "m"},`
